### PR TITLE
JSRS migration

### PR DESCRIPTION
**Pull request by gabrieloliveira41**

### CODE REVIEW

**Comment by gabrieloliveira41** [comment]
Goals verified after changes (3 out of 3 goals total):
- CommitOrder
Take another quiz
start
give me a funny quiz
start the president quiz
quiz

- Quiz
give me the funny quiz
funny quiz
the funny one

- Answer
foo bar foo bar

All tests resulted in equal intents and equal IDE simulator results after migration comparing to before the migration

**Comment by gabrieloliveira41** [comment]
on file `UpdateQuiz.js` on line 31 the IDE reported 6 errors cause the variable `ret` is not defined 
image
so i declared the variable and there were no errors left on the file
on lib `util.js` it had the same error so i did the same fix 
image
and for one of the final 2 errors 
image
i changed `var acceptedAnswer` for `let acceptedAnswer`
image
and for the final error: 
image
i that noticed quizzes is imported
image
so i created a variable to recieve `quizzes` and updated the function 
buildQuizzes before changes:
image
buildQuizzes after changes:
image

**Comment by gabrieloliveira41** [comment]
on file `UpdateQuiz.js` on line 31 the IDE reported 6 errors cause the variable `ret` is not defined 
image
so i declared the variable and there were no errors left on the file
on lib `util.js` it had the same error so i did the same fix 
image
and for one of the final 2 errors 
image
i changed `var acceptedAnswer` for `let acceptedAnswer`
image
and for the final error: 
image
i that noticed quizzes is imported
image
so i created a variable to recieve `quizzes` and updated the function 
buildQuizzes before changes:
image
buildQuizzes after changes:
image

**Comment by gabrieloliveira41** [comment]
and after that the debug tool reported an error on the file `FindQuiz.js` 
image
`FindQuiz.js` before changes:
image
since utul has a function with the same name as the import i changed the import name 
`FindQuiz.js` after changes: 
image

**Comment by gabrieloliveira41** [comment]
Hello @kangawoo44 @abapayeViv, could you review my PR, please?

File: code/lib/util.js [hunk 1]
```diff
@@ -122,30 +128,30 @@ function buildAcceptedAnswer(answer, options) {
...
-      quizzes = foundQuiz;
+      quizzValue = foundQuiz;
     }
   }
   var formattedQuizzes = [];
   //read the questions in the quiz and initialize the state
-  for (var i = 0; i < quizzes.length; i++) {
-    var quiz = quizzes[i];
+  for (var i = 0; i < quizzValue.length; i++) {
+    var quiz = quizzValue[i];
```
Comment (kangawoo44) [inline code/lib/util.js:146]: Can you test and see if this works?

```suggestion
  const foundQuiz = searchTerm ? findItems(quizzes, searchTerm) : undefined;

  var formattedQuizzes = [];
  //read the questions in the quiz and initialize the state
  for (var i = 0; i < foundQuiz.length; i++) {
    var quiz = foundQuiz[i];
```

File: code/lib/util.js [hunk 2]
```diff
@@ -226,15 +242,15 @@ function checkAliasMatch(answer, possOptionAlias, correctAlias) {
...
   return {
     answeredOption: answeredOption,
-    correct: correct
-  }
+    correct: correct,
+  };
 }
 
-module.exports = {
+export default {
```
Comment (kangawoo44) [inline code/lib/util.js]: export individually

File: code/FindQuiz.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
-var console = require("console");
-const { buildQuizzes } = require("./lib/util.js");
+import console from 'console';
+import quizzbuild from "./lib/util.js";
 
-exports.function = function(searchTerm) {
-  return buildQuizzes(searchTerm);
-};
+export default function ({ searchTerm }) {
+  return quizzbuild.buildQuizzes(searchTerm);
```
Comment (kangawoo44) [inline code/FindQuiz.js]: import { buildQuizzes } after exporting the function individually in utils.js

**Comment by kangawoo44** [comment]
@gabrieloliveira41 can you take a look at my suggestion for `utils.js`?

File: code/lib/util.js (hunk 1 above)
Comment (gabrieloliveira41) [reply code/lib/util.js:146]: with the suggested code the capsule only works when you specify the kind of quiz you want example:
"start quiz" doens't work
image
"give me a funny quiz" works
image

**Review by kangawoo44 — APPROVED** [review]
LGTM

**Review by abapayeViv — APPROVED** [review]
👍

**Pull request merged**